hnn_core/network.py

```diff
@@ -363,45 +370,116 @@ def __init__(self, times=None, gids=None, types=None):
                 if not isinstance(trial_list, list):
                     raise TypeError('%s should be a list of lists'
                                     % (arg_names[arg_idx],))
-            # Set the length of 'times' as a references and validate
+            # Set the length of 'spike_times' as a references and validate
             # uniform length
-            if arg == times:
-                n_trials = len(times)
+            if arg == spike_times:
+                n_trials = len(spike_times)
             if len(arg) != n_trials:
-                raise ValueError('times, gids, and types should be lists of '
-                                 'the same length')
-        self._times = times
-        self._gids = gids
-        self._types = types
+                raise ValueError('spike times, gids, and types should be '
+                                 'lists of the same length')
+        self._spike_times = spike_times
+        self._spike_gids = spike_gids
+        self._spike_types = spike_types
+        self._vsoma = list()
+        self._times = list()
 
     def __repr__(self):
         class_name = self.__class__.__name__
-        n_trials = len(self._times)
+        n_trials = len(self._spike_times)
         return '<%s | %d simulation trials>' % (class_name, n_trials)
 
     def __eq__(self, other):
         if not isinstance(other, Spikes):
             return NotImplemented
         # Round each time element
         times_self = [[round(time, 3) for time in trial]
-                      for trial in self._times]
+                      for trial in self._spike_times]
         times_other = [[round(time, 3) for time in trial]
-                       for trial in other._times]
+                       for trial in other._spike_times]
         return (times_self == times_other and
-                self._gids == other._gids and
-                self._types == other._types)
+                self._spike_gids == other._spike_gids and
+                self._spike_types == other._spike_types)
+
+    def __getitem__(self, gid_item):
+        """Returns a Spikes object with a copied subset filtered by gid.
```

Comment:
I'm guessing we'll revisit this soon in a future PR? So I won't look carefully into this ...

Reply:
That's the plan, the current implementation just exists to test the functionality without impacting any existing code.